cbioportal_etl/main.py

```diff
@@ -0,0 +1,277 @@
+""" 
+Wrapper script to run all scripts in ETL pipeline
+
+Run on Mgmt-Console-Dev-chopd3bprod@684194535433 EC2 instance
+
+Run a specific step, or 2 steps, or all steps:
+python3 run_pipeline.py --step [1, 2 3, or all]
+
+Steps list:
+1 - Get study metadata
+2 - Get files from manifest
+3 - Check downloaded files
+4 - Build genomic file package
+
+Example: 
+pip install -e ~/tools/kf-cbioportal-etl/
+cbio_etl_runner --help
+
+cbio_etl_runner \
+    --steps all \
+    --db-ini /path/to/db.ini \
+    --study oligo_nation \
+    --meta-config-file oligo_nation_case_meta_config.json \
+    --data-processing-config oligo_nation_data_processing_config.json \
+    --sbg-profile default \
+    --study-flag kf
+"""
+
+import argparse
+import os
+import subprocess
+import sys
+import requests
+import tarfile
+
+TOOL_DIR = os.path.dirname(os.path.abspath(__file__))
+SCRIPTS_DIR = os.path.join(TOOL_DIR, "scripts")
+CONFIGS_DIR = os.path.join(TOOL_DIR, "STUDY_CONFIGS")
+REFS_DIR = os.path.join(TOOL_DIR, "REFS")
+
+def fetch_validator_scripts():
+    repo_url = "https://github.com/cBioPortal/cbioportal/archive/refs/tags/v5.4.10.tar.gz"
+    tar_file = "v5.4.10.tar.gz"
+    extract_dir = os.path.join(TOOL_DIR, "external_scripts")
+
+    if not os.path.exists(os.path.join(extract_dir, "cbioportal-5.4.10")):
+        sys.stderr.write("Downloading cBioPortal repository...\n")
+        response = requests.get(repo_url, stream=True)
+        with open(tar_file, "wb") as f:
+            for chunk in response.iter_content(chunk_size=1024):
+                f.write(chunk)
+
+        sys.stderr.write("Extracting cBioPortal repository...\n")
+        with tarfile.open(tar_file, "r:gz") as tar:
+            tar.extractall(path=TOOL_DIR)
+        os.remove(tar_file)
+
+def resolve_config_path(provided_path, default_path):
+    """
+    Resolve the configuration path based on user input:
+    1. If no value is provided, use the default (CONFIGS_DIR/{study}_{config_name}.json)
+    2. If a file name is provided, check if it's in cwd or CONFIGS_DIR
+    3. If an absolute path is provided, use it as is.
+    """
+    if not provided_path:
+        return default_path
```

Comment:
Overall not a fan of this function having multiple responsibilities. Defaults should be handled in argument parsing. This function should solely focus on the resolving non-absolute paths.